src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -3,102 +3,203 @@
     // return AssignmentDTO with assignmentID generated by database
     @PostMapping("/assignments")
     public AssignmentDTO createAssignment(
-            @RequestBody AssignmentDTO dto) {
+            @RequestBody AssignmentDTO assignmentDTO) {
 
-        // TODO remove the following line when done
+        Section s = sectionRepository.findById(assignmentDTO.secId()).orElse(null);
+        if (s == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Section " + s.getSecId() + " not found");
+        }
 
-        return null;
+        Assignment a = new Assignment();
+        a.setSection(s);
+
+        // assignmentId generated by database
+
+        return getAssignmentDTO(assignmentDTO, a);
     }
 
-    // update assignment for a section.  Only title and dueDate may be changed.
+    // update assignment for a section. Only title and dueDate may be changed.
     // user must be instructor of the section
     // return updated AssignmentDTO
     @PutMapping("/assignments")
-    public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO dto) {
+    public AssignmentDTO updateAssignment(@RequestBody AssignmentDTO assignmentDTO) {
+
+        Assignment a = assignmentRepository.findById(assignmentDTO.id()).orElse(null);
+        if (a == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Assignment " + assignmentDTO.id() + " not found");
+        }
+        return getAssignmentDTO(assignmentDTO, a);
+    }
+
+    private AssignmentDTO getAssignmentDTO(
+        @RequestBody AssignmentDTO assignmentDTO,
+        Assignment a) {
+        a.setTitle(assignmentDTO.title());
+
+        java.sql.Date sqlDueDate;
+        try {
+            // Parse the string directly into a java.sql.Date object
+            sqlDueDate = java.sql.Date.valueOf(assignmentDTO.dueDate());
+        } catch (IllegalArgumentException e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid due date format");
+        }
+        a.setDueDate(sqlDueDate);
 
-        // TODO remove the following line when done
+        assignmentRepository.save(a);
 
-        return null;
+        return new AssignmentDTO(
+            a.getAssignmentId(),
+            a.getTitle(),
+            a.getDueDate().toString(),
+            a.getSection().getCourse().getCourseId(),
+            a.getSection().getSecId(),
+            a.getSection().getSectionNo()
+        );
     }
 
     // delete assignment for a section
-    // logged in user must be instructor of the section
+    // logged-in user must be instructor of the section
     @DeleteMapping("/assignments/{assignmentId}")
     public void deleteAssignment(@PathVariable("assignmentId") int assignmentId) {
 
-        // TODO
+        Assignment a = assignmentRepository.findByAssignmentId(assignmentId);
+        if (a == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Assignment " + assignmentId + " not found");
```

Comment:
Deletion should generally be error-free if the resource doesn't exist.

Reply:
Method changed to:
`Assignment a = assignmentRepository.findByAssignmentId(assignmentId);
        assignmentRepository.delete(a);`